File: Laboratory/Rust/functionality-testing/src/enum_size.rs
use crate::generated::enum_size::HugeEnum;
use crate::generated::enum_size::SmallEnum;
use std::mem::size_of;

#[test]
fn correct_sizes() {
    assert_eq!(size_of::<SmallEnum>(), size_of::<u8>());
    assert_eq!(size_of::<HugeEnum>(), size_of::<i64>());
}

File: Laboratory/C++/test/enum_size.cpp
#include "../gen/enum_size.hpp"
#include <cstdio>

int main() {
  if (sizeof(SmallEnum) != sizeof(uint8_t)) {
    printf("Test failed: incorrect size for HugeEnum\n");
    return 1;
  }
  if (sizeof(HugeEnum) != sizeof(int64_t)) {
    printf("Test failed: incorrect size for HugeEnum\n");
    return 1;
  }
  printf("All tests passed.\n");
  return 0;
}
